overlays/default.nix

```diff
@@ -90,7 +90,23 @@ in
     ];
   };
 
-  linuxPackages_cachyos = final.linuxPackagesFor final.linux_cachyos;
+  linuxPackages_cachyos =
+    let
+      base = final.linuxPackagesFor final.linux_cachyos;
+
+      zfs = base.zfsUnstable.overrideAttrs (pa: {
+        src =
+          final.fetchFromGitHub {
+            owner = "cachyos";
+            repo = "zfs";
+            rev = "ac18dc77f3703940682aecb442f4e58aa2c14f1a";
+            hash = "sha256-wrMZYENs4hmrHXcSN4kYgntaDDs5IwOMeWWqUKortbw=";
+          };
+        meta = pa.meta // { broken = false; };
+        patches = [ ];
+      });
+    in
+    base.extend(_: _: { zfs = zfs; zfsStable = zfs; zfsUnstable = zfs; });
```

Comment:
I can finally sleep in peace, because after hours debugging why replacing ZFS using `// { zfs = ... }` would not really change anything, I finally discovered that for `lib.makeExtensible` (which `linuxPackagesFor` uses) you have to use `.extend`.